modules/dxp/apps/segments/segments-asah-connector-test/src/testIntegration/java/com/liferay/segments/asah/connector/internal/model/listener/test/LayoutModelListenerTest.java

```diff
@@ -0,0 +1,190 @@
+import org.junit.runner.RunWith;
+
+/**
+ * @author Mikel Lorza
+ */
+@RunWith(Arquillian.class)
+public class LayoutModelListenerTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			_group.getCompanyId(), _group.getGroupId(),
+			TestPropsValues.getUserId());
+	}
+
+	@Test
+	public void testOnBeforeUpdate() throws Exception {
+		Layout layout1 = LayoutTestUtil.addTypeContentLayout(_group);
+
+		long defaultSegmentsExperienceId =
+			_segmentsExperienceLocalService.fetchDefaultSegmentsExperienceId(
+				layout1.getPlid());
+
+		SegmentsExperiment segmentsExperiment =
+			_segmentsExperimentLocalService.addSegmentsExperiment(
+				defaultSegmentsExperienceId,
+				_portal.getClassNameId(Layout.class), layout1.getPlid(),
+				"AB test", "A/B test description",
+				SegmentsExperimentConstants.Goal.BOUNCE_RATE.getLabel(),
+				StringPool.BLANK, _serviceContext);
+
+		Object asahSegmentsExperimentProcessor =
+			ReflectionTestUtil.getFieldValue(
+				_modelListener, "_asahSegmentsExperimentProcessor");
+
+		Object asahFaroBackendClient = ReflectionTestUtil.getFieldValue(
+			asahSegmentsExperimentProcessor, "_asahFaroBackendClient");
+
+		ReflectionTestUtil.setFieldValue(
+			asahFaroBackendClient, "_http",
+			MockHttpUtil.geHttp(
+				Collections.singletonMap(
+					"/api/1.0/experiments/" +
+						segmentsExperiment.getSegmentsExperienceId(),
+					() -> "{}")));
+
+		SegmentsExperience segmentsExperience =
+			_segmentsExperienceLocalService.addSegmentsExperience(
+				TestPropsValues.getUserId(), _group.getGroupId(), 0,
+				_portal.getClassNameId(Layout.class), layout1.getPlid(),
+				Collections.singletonMap(
+					LocaleUtil.getSiteDefault(), "Variant 1"),
+				false, new UnicodeProperties(true), _serviceContext);
+
+		_segmentsExperimentRelService.addSegmentsExperimentRel(
+			segmentsExperiment.getSegmentsExperimentId(),
+			segmentsExperience.getSegmentsExperienceId(), _serviceContext);
+
+		try (CompanyConfigurationTemporarySwapper
+				companyConfigurationTemporarySwapper =
+					new CompanyConfigurationTemporarySwapper(
+						TestPropsValues.getCompanyId(),
+						AnalyticsConfiguration.class.getName(),
+						HashMapDictionaryBuilder.<String, Object>put(
+							"liferayAnalyticsDataSourceId",
+							RandomTestUtil.randomLong()
+						).put(
+							"liferayAnalyticsEnableAllGroupIds", true
+						).put(
+							"liferayAnalyticsFaroBackendSecuritySignature",
+							RandomTestUtil.randomString()
+						).put(
+							"liferayAnalyticsFaroBackendURL",
+							RandomTestUtil.randomString()
+						).build(),
+						SettingsFactoryUtil.getSettingsFactory())) {
+
+			String title = "Layout title modified";
+
+			layout1.setTitleMap(
```

Comment:
It is not a good practice to mutate a variable, what do you think about calling the method updateLayout with all the parameters?

```
	public Layout updateLayout(
			long groupId, boolean privateLayout, long layoutId,
			long parentLayoutId, Map<Locale, String> nameMap,
			Map<Locale, String> titleMap, Map<Locale, String> descriptionMap,
			Map<Locale, String> keywordsMap, Map<Locale, String> robotsMap,
			String type, boolean hidden, Map<Locale, String> friendlyURLMap,
			boolean hasIconImage, byte[] iconBytes, long styleBookEntryId,
			long faviconFileEntryId, long masterLayoutPlid,
			ServiceContext serviceContext)
		throws PortalException 
```

Reply:
This class has been deleted and I added some unit tests of AsahFaroBackendClientImplTest class.